tools/chpl-language-server/src/symbol_signature.py

```diff
@@ -17,23 +17,43 @@
 # limitations under the License.
 #
 
-from typing import Optional, Union
+from typing import List, Optional, Union
 import chapel
 
 
-# TODO: for now, just text based. but should resolve generic types
-def get_symbol_signature(node: chapel.AstNode) -> str:
+class SymbolSignature:
+    def __init__(self, node: chapel.AstNode):
+        self.node = node
+        self._signature = get_symbol_signature(self.node, False, False)
+
+    def compute_type(self):
+        """
+        if types aren't textually present, try to resolve them
+
+        Note: this is a temporary method that will go away when the resolver is fully online
+        """
+        self._signature = get_symbol_signature(self.node, False, True)
+
+    def compute_value(self):
+        """evaluate expressions"""
+        self._signature = get_symbol_signature(self.node, True, True)
+
+    def __str__(self) -> str:
+        return self._signature
```

Comment:
Considering passing `False` and `True` here just append stuff to the result of `get_symbol_signature`, consider the following:


```Python
class SymbolSignature:
    def __init__(self, node: chapel.AstNode):
        self.node = node
        self._signature = get_symbol_signature(self.node)
        self._type = ""
        self._value = ""

    def compute_type(self):
        """
        if types aren't textually present, try to resolve them

        Note: this is a temporary method that will go away when the resolver is fully online
        """
        self._type = f": {get_symbol_type(self.node)}"

    def compute_value(self):
        """evaluate expressions"""
        self._value = f"= {get_symbol_value(self.node)}"
     
    def __str__(self) -> str:
        return self._signature + self._type + self._value
```

Reply:
the above might need tweaks to handle None, but you get the idea.